### PR TITLE
Revert of PR #10012 -  10012 update apache ruleset

### PR DESCRIPTION
**Pull request by 72nomada**
|Related issue|
|---|
|#10976 |

## Description

This PR aims to revert PR #10064

Reason to revert
- Parent decoder is too generic. 

Changes done
Note: no decoder, rule, or test behavior was changed. 

- Decoders 
   - Copyright and product info header added 
   - Indent updated 
  
- Rules 
  - Description tag syntax
  - XML tag Indent 
  - Group tag order 
   - Copyright and product info header added 

- Tests
   - Test description updated
   - Copyright and product info header added 

### Syntax

- [X] 1.a Rule tags order must be compliant.
- [X] 1.b Decoder tags order must be compliant.
- [X] 1.c XML blocks must as compact as possible.
- [X] 1.d Only one empty line between rule/decoder and the next rule/decoder.
- [X] 1.e Decoder extracted fields names must use '_' whether a space is needed.
- [X] 1.f Decoder name must use '-' whether a space is needed.
 
### Grammar

- [X] 2.a Grammar quality.
- [X] 2.b Similar phrases must keep tenses and expressions.
- [X] 2.c Grammar basic rules like capitalization, punctuation marks, sentence construction, and others.

### Semantic

- ~~3.a New decoder, rule, or test are written in the correct file and grouped correctly inside the file.~~
- ~~3.b Group similar rules under same ID's group.~~
- [X] 3.c:
     - ~~[X] 3.c1 Find and reuse group name before creating a new one.~~
     - ~~[X] 3.c2 Include a new group definition in a PR comment.~~
     - [X] 3.c3 Any group name must use '_' whether it does need a space char.~~
     - [X] 3.c4 The groups inside a tag must be sorted in alphabetical order.
- ~~3.d Rule level should be compliant with the [documentation](https://documentation.wazuh.com/current/user-manual/ruleset/rules-classification.html).~~

### Unit testing

- ~~4.a A metadata block at the beginning of the .ini file describing software, version, and logs source.~~
- ~~4.b Each new rule must have at least one test entry in the correct .ini file.~~
- [x] 4.c Runtest.py must pass and must include the results in raw format here.

<details><summary>All rules and decoders test.</summary>
<p>

```
python runtests.py
- [ File = ./tests/SonicWall.ini ] ---------
........

- [ File = ./tests/apache.ini ] ---------
..................

- [ File = ./tests/api.ini ] ---------
............................

- [ File = ./tests/apparmor.ini ] ---------
.....

- [ File = ./tests/arbor.ini ] ---------
..

- [ File = ./tests/audit_lateral.ini ] ---------
......

- [ File = ./tests/audit_recon.ini ] ---------
..

- [ File = ./tests/audit_scp.ini ] ---------
.

- [ File = ./tests/auditd.ini ] ---------
...

- [ File = ./tests/aws_s3_access.ini ] ---------
.......

- [ File = ./tests/checkpoint_smart1.ini ] ---------
..................

- [ File = ./tests/cimserver.ini ] ---------
..

- [ File = ./tests/cisco_asa.ini ] ---------
........................................................................................

- [ File = ./tests/cisco_ftd.ini ] ---------
..........................................

- [ File = ./tests/cisco_ios.ini ] ---------
.................

- [ File = ./tests/cloudlfare-waf.ini ] ---------
.............

- [ File = ./tests/cpanel.ini ] ---------
.......

- [ File = ./tests/cylance.ini ] ---------
.......

- [ File = ./tests/doas.ini ] ---------
....

- [ File = ./tests/dovecot.ini ] ---------
...............

- [ File = ./tests/dropbear.ini ] ---------
...

- [ File = ./tests/eset.ini ] ---------
........

- [ File = ./tests/exchange.ini ] ---------
..

- [ File = ./tests/exim.ini ] ---------
.......

- [ File = ./tests/f5_big_ip.ini ] ---------
...........................................

- [ File = ./tests/fireeye.ini ] ---------
...

- [ File = ./tests/firewalld.ini ] ---------
..

- [ File = ./tests/fortiauth.ini ] ---------
....

- [ File = ./tests/fortigate.ini ] ---------
........................................

- [ File = ./tests/freepbx.ini ] ---------
......

- [ File = ./tests/gcp.ini ] ---------
...........

- [ File = ./tests/github.ini ] ---------
....................................................................................................................................................................................................................................................................................................................................

- [ File = ./tests/gitlab.ini ] ---------
.........................

- [ File = ./tests/glpi.ini ] ---------
...

- [ File = ./tests/huawei_usg.ini ] ---------
...

- [ File = ./tests/icinga.ini ] ---------
....

- [ File = ./tests/iptables.ini ] ---------
........

- [ File = ./tests/junos.ini ] ---------
...

- [ File = ./tests/kernel_usb.ini ] ---------
......

- [ File = ./tests/mailscanner.ini ] ---------
.

- [ File = ./tests/mcafee_epo.ini ] ---------
.

- [ File = ./tests/modsecurity.ini ] ---------
......

- [ File = ./tests/named.ini ] ---------
.....

- [ File = ./tests/netscreen.ini ] ---------
....

- [ File = ./tests/nextcloud.ini ] ---------
.......

- [ File = ./tests/nginx.ini ] ---------
............

- [ File = ./tests/office365.ini ] ---------
................................................................................................................................

- [ File = ./tests/openldap.ini ] ---------
.........

- [ File = ./tests/opensmtpd.ini ] ---------
.......

- [ File = ./tests/openvpn_ldap.ini ] ---------
..

- [ File = ./tests/oscap.ini ] ---------
................................

- [ File = ./tests/ossec.ini ] ---------
.....

- [ File = ./tests/owlh.ini ] ---------
....

- [ File = ./tests/paloalto.ini ] ---------
................

- [ File = ./tests/pam.ini ] ---------
.....

- [ File = ./tests/panda_paps.ini ] ---------
........

- [ File = ./tests/pfsense.ini ] ---------
..

- [ File = ./tests/pix.ini ] ---------
......................

- [ File = ./tests/postfix.ini ] ---------
..

- [ File = ./tests/proftpd.ini ] ---------
.......

- [ File = ./tests/rsh.ini ] ---------
..

- [ File = ./tests/samba.ini ] ---------
....

- [ File = ./tests/sophos-utm-firewall.ini ] ---------
......

- [ File = ./tests/sophos.ini ] ---------
........

- [ File = ./tests/sophos_fw.ini ] ---------
..........

- [ File = ./tests/squid_rules.ini ] ---------
..

- [ File = ./tests/sshd.ini ] ---------
...........................................

- [ File = ./tests/su.ini ] ---------
.....

- [ File = ./tests/sudo.ini ] ---------
........

- [ File = ./tests/syslog.ini ] ---------
......

- [ File = ./tests/sysmon.ini ] ---------
...

- [ File = ./tests/systemd.ini ] ---------
..

- [ File = ./tests/trendmicro-cloud-one.ini ] ---------
.................

- [ File = ./tests/unbound.ini ] ---------


- [ File = ./tests/vsftpd.ini ] ---------
....

- [ File = ./tests/vuln_detector.ini ] ---------
..

- [ File = ./tests/web_appsec.ini ] ---------
...............................

- [ File = ./tests/web_rules.ini ] ---------
..........

|Component |  Tested  |  Total   | Coverage |
| -------- | -------- | -------- | -------- |
|  Rules   |   982    |   4174   |  23.53%  |
| Decoders |    77    |   170    |  45.29%  |

|          File           |  Passed  |  Failed  |  Status  |
|        --------         | -------- | -------- | -------- |
|./tests/sysmon.ini       |    3     |    0     |   ✅    |
|./tests/proftpd.ini      |    7     |    0     |   ✅    |
|./tests/glpi.ini         |    3     |    0     |   ✅    |
|./tests/dovecot.ini      |    15    |    0     |   ✅    |
|./tests/exim.ini         |    7     |    0     |   ✅    |
|./tests/gcp.ini          |    11    |    0     |   ✅    |
|./tests/pix.ini          |    22    |    0     |   ✅    |
|./tests/cpanel.ini       |    7     |    0     |   ✅    |
|./tests/sophos.ini       |    8     |    0     |   ✅    |
|./tests/syslog.ini       |    6     |    0     |   ✅    |
|./tests/checkpoint_smart1.ini|    18    |    0     |   ✅    |
|./tests/nextcloud.ini    |    7     |    0     |   ✅    |
|./tests/iptables.ini     |    8     |    0     |   ✅    |
|./tests/samba.ini        |    4     |    0     |   ✅    |
|./tests/cisco_asa.ini    |    88    |    0     |   ✅    |
|./tests/dropbear.ini     |    3     |    0     |   ✅    |
|./tests/systemd.ini      |    2     |    0     |   ✅    |
|./tests/SonicWall.ini    |    8     |    0     |   ✅    |
|./tests/web_appsec.ini   |    31    |    0     |   ✅    |
|./tests/squid_rules.ini  |    2     |    0     |   ✅    |
|./tests/firewalld.ini    |    2     |    0     |   ✅    |
|./tests/mailscanner.ini  |    1     |    0     |   ✅    |
|./tests/owlh.ini         |    4     |    0     |   ✅    |
|./tests/fortiauth.ini    |    4     |    0     |   ✅    |
|./tests/cisco_ios.ini    |    17    |    0     |   ✅    |
|./tests/vuln_detector.ini|    2     |    0     |   ✅    |
|./tests/cloudlfare-waf.ini|    13    |    0     |   ✅    |
|./tests/openldap.ini     |    9     |    0     |   ✅    |
|./tests/kernel_usb.ini   |    6     |    0     |   ✅    |
|./tests/auditd.ini       |    3     |    0     |   ✅    |
|./tests/f5_big_ip.ini    |    43    |    0     |   ✅    |
|./tests/mcafee_epo.ini   |    1     |    0     |   ✅    |
|./tests/opensmtpd.ini    |    7     |    0     |   ✅    |
|./tests/sudo.ini         |    8     |    0     |   ✅    |
|./tests/unbound.ini      |    0     |    0     |   ✅    |
|./tests/icinga.ini       |    4     |    0     |   ✅    |
|./tests/netscreen.ini    |    4     |    0     |   ✅    |
|./tests/doas.ini         |    4     |    0     |   ✅    |
|./tests/pam.ini          |    5     |    0     |   ✅    |
|./tests/arbor.ini        |    2     |    0     |   ✅    |
|./tests/office365.ini    |   128    |    0     |   ✅    |
|./tests/paloalto.ini     |    16    |    0     |   ✅    |
|./tests/cylance.ini      |    7     |    0     |   ✅    |
|./tests/fortigate.ini    |    40    |    0     |   ✅    |
|./tests/ossec.ini        |    5     |    0     |   ✅    |
|./tests/web_rules.ini    |    10    |    0     |   ✅    |
|./tests/oscap.ini        |    32    |    0     |   ✅    |
|./tests/exchange.ini     |    2     |    0     |   ✅    |
|./tests/audit_lateral.ini|    6     |    0     |   ✅    |
|./tests/cisco_ftd.ini    |    42    |    0     |   ✅    |
|./tests/sophos_fw.ini    |    10    |    0     |   ✅    |
|./tests/github.ini       |   324    |    0     |   ✅    |
|./tests/postfix.ini      |    2     |    0     |   ✅    |
|./tests/audit_scp.ini    |    1     |    0     |   ✅    |
|./tests/api.ini          |    28    |    0     |   ✅    |
|./tests/sophos-utm-firewall.ini|    6     |    0     |   ✅    |
|./tests/rsh.ini          |    2     |    0     |   ✅    |
|./tests/apparmor.ini     |    5     |    0     |   ✅    |
|./tests/nginx.ini        |    12    |    0     |   ✅    |
|./tests/named.ini        |    5     |    0     |   ✅    |
|./tests/junos.ini        |    3     |    0     |   ✅    |
|./tests/openvpn_ldap.ini |    2     |    0     |   ✅    |
|./tests/cimserver.ini    |    2     |    0     |   ✅    |
|./tests/vsftpd.ini       |    4     |    0     |   ✅    |
|./tests/freepbx.ini      |    6     |    0     |   ✅    |
|./tests/sshd.ini         |    43    |    0     |   ✅    |
|./tests/modsecurity.ini  |    6     |    0     |   ✅    |
|./tests/pfsense.ini      |    2     |    0     |   ✅    |
|./tests/trendmicro-cloud-one.ini|    17    |    0     |   ✅    |
|./tests/huawei_usg.ini   |    3     |    0     |   ✅    |
|./tests/su.ini           |    5     |    0     |   ✅    |
|./tests/apache.ini       |    18    |    0     |   ✅    |
|./tests/gitlab.ini       |    25    |    0     |   ✅    |
|./tests/aws_s3_access.ini|    7     |    0     |   ✅    |
|./tests/eset.ini         |    8     |    0     |   ✅    |
|./tests/fireeye.ini      |    3     |    0     |   ✅    |
|./tests/panda_paps.ini   |    8     |    0     |   ✅    |
|./tests/audit_recon.ini  |    2     |    0     |   ✅    |
```
</p>
</details>

- ~~4.d New CDB lists must include the proper test entry in the correct .ini file.~~

### E2E testing

- ~~5.a Logs for new or modified decoder/rules sent to a manager running with this PR ruleset appear in Kibana.~~ 
- ~~5.b There are not affected or broken visualizations on Kibana.~~
- ~~5.c New or modified items can be seen correctly using APP ruleset navigation.~~

### Elasticsearch Template

- ~~6.a Known fields with output format managed and usually used for searching are included in template array index.query.default_field.~~
- ~~6.b The new field with the correct date format is stored as a "date" type field in the template.~~
- ~~6.c The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template.~~
- ~~6.d Known fields with output format managed and usually used for searching are included in the template.~~

### Stoppers

- ~~7.a No previous rule ID changes without triple check.~~
- ~~7.b No previous decoder name changes without triple check.~~
- ~~7.c No previous file name changes without triple check.~~
- ~~7.d No previous test changes its 'rule' field value without triple check.~~

### Others

- ~~8.a Each file has the correct copyright block.~~
- ~~8.b The copyright block doesn't have "Author" only "Created by Wazuh". To include an "Author" request triple check.~~
- ~~8.c The copyright block doesn't use "-".~~
- ~~8.d The rule files don't have any sample log.~~
- ~~8.e The decoder file has sample logs next to the decoder that matches that log.~~
- ~~8.f The decoder and rule files have information about software, version, and any helpful information.~~
- ~~8.g The PR has a single commit with CHANGELOG changes in the correct format.~~
- ~~8.h The new rule ID is not in use.~~
- ~~8.i The new rule ID must be in the defined IDs range.~~
- ~~8.j New rules ID range must be verified with a triple check and noted in the rules ID document.~~
- ~~8.k The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template.~~

### CODE REVIEW

**Review by xrisbarney — APPROVED** [review]
LGTM.

File: ruleset/decoders/0025-apache_decoders.xml [hunk 1]
```diff
@@ -1,86 +1,68 @@
...
-    <parent>apache-errorlog</parent>
-    <regex offset="after_parent" type="pcre2">\s\w+:\s([^"]+)"([^"]+)":\s([^\[]+)\[RemoteIP:\s([^\]]+)]</regex>
-    <order>apache.error_message, user, apache.error_reason, apache.remote_ip</order>
-</decoder>
+<decoder name="apache-errorlog-ip">
+  <parent>apache-errorlog</parent>
+  <prematch offset="after_parent">[client</prematch>
+  <regex offset="after_prematch">^ (\S+):(\d+)] |^ (\S+)] </regex>
+  <order>srcip,srcport</order>
+</decoder>
```
Comment (fabamatic) [inline ruleset/decoders/0025-apache_decoders.xml:68]: Missing new line at end
```suggestion
</decoder>

```

File: ruleset/rules/0250-apache_rules.xml [hunk 2]
```diff
@@ -329,7 +317,7 @@
     <mitre>
       <id>T1083</id>
     </mitre>
-    <group>modsecurity,gpg13_10.1,</group>
+    <group>gpg13_10.1,modsecurity,</group>
   </rule>
 
-</group>
+</group>
```
Comment (fabamatic) [inline ruleset/rules/0250-apache_rules.xml:323]: Missing newline at the end
```suggestion
</group>

```

File: ruleset/testing/tests/apache.ini [hunk 3]
```diff
@@ -1,79 +1,71 @@
...
 alert = 0
 decoder = apache-errorlog
 
-[PHP Notices in Apache 2.4 errorlog]
+[Apache: PHP Notices in Apache 2.4 errorlog]
 log 1 pass = [Sun Nov 23 18:49:01.713508 2014] [:error] [pid 15816] [client 141.8.147.9:51507] PHP Notice:  A non well formed numeric value encountered in /path/to/file.php on line 123
 rule = 30318
 alert = 5
-decoder = apache-errorlog
+decoder = apache-errorlog
```
Comment (fabamatic) [inline ruleset/testing/tests/apache.ini:71]: Missing newline at the end
```suggestion
decoder = apache-errorlog

```

File: ruleset/rules/0250-apache_rules.xml [hunk 4]
```diff
@@ -1,92 +1,89 @@
...
     <match>^[warn] </match>
-    <description>Apache warn messages grouped.</description>
+    <description>Apache: warn messages grouped.</description>
   </rule>
 
   <rule id="30103" level="0">
     <if_sid>30100</if_sid>
     <match>^[notice] </match>
-    <description>Apache notice messages grouped.</description>
+    <description>Apache: notice messages grouped.</description>
```
Comment (Spades0) [inline ruleset/rules/0250-apache_rules.xml:31]: Apache: Notice messages grouped

File: ruleset/rules/0250-apache_rules.xml [hunk 5]
```diff
@@ -1,92 +1,89 @@
...
   </rule>
 
   <rule id="30104" level="12">
     <if_sid>30103</if_sid>
     <match>exit signal Segmentation Fault</match>
-    <description>Apache: segmentation fault.</description>
     <info type="link">http://www.securityfocus.com/infocus/1633</info>
-    <group>service_availability,pci_dss_6.5.2,pci_dss_6.6,gpg13_4.14,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SC.5,nist_800_53_SI.3,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,</group>
+    <description>Apache: segmentation fault.</description>
```
Comment (Spades0) [inline ruleset/rules/0250-apache_rules.xml:38]: Apache: Segmentation fault

**Comment by Spades0** [comment]
LGTM

**Review by Spades0 — APPROVED** [review]
LGTM

File: ruleset/rules/0250-apache_rules.xml [hunk 6]
```diff
@@ -1,92 +1,89 @@
...
+  Copyright (C) 2015-2021, Wazuh Inc.
+-->
+
+<!-- 
+  Rules for:
+    Apache HTTP Server ID: 30100 - 30500
 -->
 
 <group name="apache,web,">
   <rule id="30100" level="0">
```
Comment (juliancnn) [inline ruleset/rules/0250-apache_rules.xml:11]: Add  a line break here
```suggestion

  <rule id="30100" level="0">
```

File: ruleset/rules/0250-apache_rules.xml [hunk 7]
```diff
@@ -95,7 +92,7 @@
     <match>failed to open stream: No such file or directory|</match>
     <match>Failed opening </match>
     <description>Apache: Attempt to access an non-existent file (those are reported on the access.log).</description>
-    <group>unknown_resource,pci_dss_10.2.1,pci_dss_10.2.4,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
+    <group>gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,pci_dss_10.2.1,pci_dss_10.2.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,unknown_resource,</group>
   </rule>
 
   <!-- [Tue Mar 07 12:05:15 2006] [error] [client 200.206.165.91] Invalid URI in request %3Bi%3A3%3Bi%3A0%3B%7D; usercookie[password]=d6ed9e1750d0b2aba6b3311cbec087d8; 45befd35f8a0f47b89ed8831f892b8dc=167c4e46a940cd2570b952eea527b27a; PHPSESSID=616hjdg7kj9bln37efsv7vt7g3
```
Comment (nmkoremblum) [inline ruleset/rules/0250-apache_rules.xml:98]:
```suggestion
```

**Review by nmkoremblum — APPROVED** [review]
LGTM :)

**Comment by juliancnn** [comment]
LGTM :)

**Review by juliancnn — APPROVED** [review]

**Pull request merged**